src/Database/LSMTree/Internal/Serialise/RawBytes.hs

```diff
@@ -143,8 +142,17 @@ toWord32 x# = byteSwap32 (W32# x#)
 sizeofRawBytes :: RawBytes -> Int
 sizeofRawBytes (RawBytes pvec) = P.length pvec
 
+{-# INLINE append #-}
+append :: RawBytes -> RawBytes -> RawBytes
+append = coerce (P.++)
+
+{-# INLINE concat #-}
 concat :: [RawBytes] -> RawBytes
-concat rbs = RawBytes (P.concat (fmap toVector rbs))
+concat = coerce P.concat
+
+{-# INLINE take #-}
+take :: Int -> RawBytes -> RawBytes
+take = coerce P.take
```

Comment:
Maybe we should just use `P.Vector` functions everywhere and use `coerce` when needed, instead of defining wrapper functions

Reply:
I'm happy enough with it being a proper abstraction, rather than effectively a type alias.